tools/data_manipulation/xarray/xarray_mapplot.xml

```diff
@@ -0,0 +1,204 @@
+    #if $range!=""
+         --range="$range"
+    #end if
+    #if $xlim!=""
+         --xlim="$xlim"
+    #end if
+    #if $ylim!=""
+         --ylim="$ylim"
+    #end if
+    #if $shift!=""
+         --shift
+    #end if
+           --verbose &&
+    mv *.png output_dir
+    ]]></command>
+    <inputs>
+        <param type="data" name="input" label="Input netcdf file" format="netcdf"/>
+        <param type="data" label="Tabular of variables" name="var_tab" format="tabular" help="Select the tabular file which summarize the available variables and dimensions."/>
+
+        <param name="var" type="select" label="Choose the variable to extract">
+            <options from_dataset="var_tab">
+                <column name="name" index="0"/>
+                <column name="value" index="0"/>
+            </options>
+        </param>
+        <param name="lat_dim" type="select" label="Name of latitude coordinate" >
+            <options from_dataset="var_tab">
+                <column name="value" index="0"/>
+            </options>
+        </param>
+        <param name="lon_dim" type="select" label="Name of longitude coordinate" >
+            <options from_dataset="var_tab">
+                <column name="value" index="0"/>
+            </options>
+        </param>
+        <param name="time_idx" type="text" value="" label="List of time indices such as 0,1,... (optional)" />
+        <param name="xlim" type="text" value="" label="longitudes values 'lonW,lonE' for limited geographical area (optional and only available with some projections)" />
+        <param name="ylim" type="text" value="" label="latitudes values 'latS,latN' for limited geographical area (optional and only available with some projections)" />
+        <param name="shift" type="select" display="radio" label="Shift longitudes [0,360] --> [-180,180]"> 
+             <option value="" selected="true">No</option>
+             <option value="Yes">Yes</option>
+        </param>
+        <param name="range" type="text" value="" label="Range of values for plotting e.g. minimum value abd maximum value (minval,maxval) (optional)" />
+        <param name="threshold" type="text" value="" label="Do not plot values below this threshold (optional)" />
+        <param name="borders" type="text" value="" label="Add country borders with alpha value [0-1] (optional)" />
+        <param name="coastline" type="text" value="" label="Add coastline with alpha value [0-1] (optional)" />
+        <param name="ocean" type="text" value="" label="Add ocean with alpha value [0-1] (optional)" />
+        <param name="land" type="text" value="" label="Add land with alpha value [0-1] (optional)" />
+        <param name="title" type="text" value="" label="Specify plot title (optional)" />
+        <param name="colorbar_label" type="text" value="" label="Set a label for colormap (optional)" />
+        <param name="cmap" type="text" value="" label="Specify which colormap to use for plotting (optional)" />
```

Comment:
do you know type="color" ?

https://docs.galaxyproject.org/en/latest/dev/schema.html#color

If you can not use it I think a better way is to have a long select box

Reply:
No but here we need a colormap (such as matplotlib colormap or cmcrameri)